### ISSUE TITLE
Purchases: Fix Purchases page that can't load for sites not belonging to current user

### ISSUE THREAD

**Issue by stephanethomas** [Automattic/wp-calypso#1870]
We should better handle the case of users who purchased an upgrade for a site they don't own. In this scenario, the `Purchases` page fails to load and displays a loading placeholder:

![screenshot](https://cloud.githubusercontent.com/assets/594356/11918285/b4598ea0-a72a-11e5-8063-2dd8eb6975c7.png)

The following warning is also displayed in the console:

```
Warning: Failed propType: Required prop `selectedSite` was not specified in `ManagePurchase`. Check the render method of `StoreConnection`.
```

Despite the fact the list of sites being not empty, it does not contain the site where the purchase was done [which leads](https://github.com/Automattic/wp-calypso/blob/0d7f8635787d5deb666e42b782d7c22fc48347c6/client/components/data/purchases/manage-purchase/index.jsx#L59) to the selected site being `undefined`.

**Comment by fabianapsimoes** [comment]
@stephanethomas Looks like this is a duplicate of https://github.com/Automattic/wp-calypso/issues/3141. Can you confirm?

**Comment by stephanethomas** [comment]
I was able to reproduce this issue by:

1. Logging as user *#*1 and select blog *#*A
2. Adding user *#*2 to blog *#*A as _contributor_ (or anything that is not _follower_)
3. Logging as user *#*2 and select blog *#*A
4. Purchasing a plan for blog *#*A
5. Checking that the corresponding `Manage Purchase` page displays correctly
6. Logging as user *#*1 and select blog *#*A
7. Remove user *#*2 from the list of contributors
8. Logging as user *#*2
9. Checking that the `Purchases` page displays as previously
10. Checking that the `Manage Purchase` for the plan displays a placeholder

I would say that this issue describes a subset of the problem mentioned in https://github.com/Automattic/wp-calypso/issues/3141. On the other hand @BandonRandon mentioned that users with any role different from _Administrator_ wouldn't be able to manage their upgrade whereas I confirmed that I could see the `Manage Purchase` page in that case.

**Comment by fabianapsimoes** [comment]
This will be fixed as part of #4067.
